src/main/java/org/squiddev/plethora/integration/vanilla/DisableAI.java

```diff
@@ -41,20 +43,19 @@ public void readNBT(Capability<IDisableAIHandler> capability, IDisableAIHandler
 		}, DefaultDisableAI::new);
 	}
 
-	public static void maybeClear(EntityLiving entity) {
+	/**
+	 * Check if this entity should have its AI disabled and if so, ensure that our obedience AI is inserted.
+	 * @param entity
+	 */
+	public static void maybePossess(EntityLiving entity) {
 		IDisableAIHandler disableAI = entity.getCapability(DISABLE_AI_CAPABILITY, null);
 		if (disableAI != null && disableAI.isDisabled()) {
-			clearTasks(entity.tasks);
-			clearTasks(entity.targetTasks);
-		}
-	}
-
-	private static void clearTasks(EntityAITasks tasks) {
-		if (tasks.taskEntries.isEmpty()) return;
-
-		List<EntityAITasks.EntityAITaskEntry> backup = Lists.newArrayList(tasks.taskEntries);
-		for (EntityAITasks.EntityAITaskEntry entry : backup) {
-			tasks.removeTask(entry.action);
+			// Check that our obedience AI task is added.
+			// We don't remove this AI as it watches the Disable AI Capability for changes.
+			Optional<EntityAITasks.EntityAITaskEntry> obedient = entity.tasks.taskEntries.stream().filter(it -> it.action instanceof AIPlethoraObedience).findFirst();
```

Comment:
Probably easier just to do:
```java
for (EntityAITasks.EntityAITaskEntry entry : entity.tasks.taskEntries) {
    if (entry.action instanceof AIPlethoraObedience) return;
}
entity.tasks.addTask(Integer.MIN_VALUE, new AIPlethoraObedience(entity));
```
Same number of lines of code, but without the insanely long line.

Alternatively:

```java
if (entity.tasks.taskEntries.noneMatch(x -> x.action instanceof AIPlethoraObedience)) {
    entity.tasks.addTask(Integer.MIN_VALUE, new AIPlethoraObedience(entity));
}
```

Reply:
I like the last one.

Resolved with https://github.com/SquidDev-CC/plethora/pull/218/commits/f0770f74df3805bac3f68c99ecf669855cae08e9